docs/adr/adr_authorization-specification.md

```diff
@@ -0,0 +1,177 @@
+An RBP refers to a resource in the database, and the methods that the user who was granted the RBP can execute. These methods correspond to database operations:
+
+- read: `SELECT`
+- create: `INSERT`
+- edit: `UPDATE`
+- write: `INSERT`, `UPDATE`
+- delete: `DELETE`
+- assign: `INSERT` into cross-reference table
+
+The naming convention for RBP is a singular noun (the resource; multi-word nouns concatenated by underscore), and a method name, separated by colon: `user:edit`, `beneficiary:create`, `tag_relation:read`, etc.
+
+Every ABP comprises one or more RBP, e.g. the ABP `manage_tags` stands for `tag:write`, `stock:read`, `tag_relation:read`, and `beneficiary:read`.
+
+The mapping of usergroup to ABPs, and ABP to RBPs is listed in [this document](https://docs.google.com/spreadsheets/d/1W4YWcc59wUFUWgReumdH6DQ4zU7JcTgvf6WEbdqaGHQ/edit#gid=0).
+
+### User management in Auth0
+
+[Auth0](https://auth0.com) is a service for managing user authentication and authorization. It serves as single source of truth. An authenticated user gets issued a JSON Web Token (JWT) holding their authorization information. For signing the JWT we use the RS256 algorithm: the token will be signed with our private signing key and can be verified using our public signing key.
+
+#### Reasons to use Auth0
+
+Cf. [related ADR](./docs/adr/adr_auth0.md)
+
+#### User entity in Auth0
+
+Any user registered for boxtribute has their authorization data (`app_metadata`) stored in the Auth0 database. The user attributes are:
+
+- `is_god`: `"1"` for god user, `"0"` otherwise
+- `usergroup_id`: the ID of the usergroup the user belongs to
+- `base_ids`: a list of base IDs that the user has access to
+- `organisation_id`: the ID of the organisation the user belongs to
+
+During registration, the user gets assigned a role, indicating their usergroup and the bases they belong to. The role is named like `base_1_coordinator`.
+
+From the user's role the ABPs are derived via a script (?).
+
+When the user has successfully logged in, an Auth0 post-login action script runs. The script creates a JWT with the content derived from user authorization data and their role.
+Most importantly the script creates base-specific RBPs for the current user (see below about their format).
+
+#### Specification of custom JWT
+
+The JWT contains standard and customs fields.
+
+Field name | Kind | Description | Usage
+:--- | :--- | :--- | :---
+`iss` | standard | Name of token issuer | JWT decoding
+`aud` | standard | Name of token audience | JWT decoding
+`iat` | standard | Unix timestamp of issuing datetime | -
+`exp` | standard | Unix timestamp of expiration datetime | JWT decoding
+`azp` | standard | ID of client through which the JWT was requested | traceability of application used for authentication
+`gty` | standard | Grant type | -
+`sub` | standard | User ID | see below
```

Comment:
I do not think the use of "see below"in this table is helpful. It should refer to specific section of the document.